changelog.d/14191.doc

```diff
@@ -0,0 +1 @@
+Update docstrings of `SynapseError` and `FederationError` to bettter describe what they are used for and the effects of using them are.
```

Comment:
Minor? Is `.doc` possibly for user-visible documentation changes rather than internal docstrings? (No strong opinions here.)

Reply:
Probably but they are experienced by Synapse admins in their logs. This gives at least some context even if it's in the code.

I'm going to opt to leave it until someone has a stronger opinion.